packages/kbn-check-saved-objects-cli/src/commands/run_check_saved_objects_cli.ts

```diff
@@ -7,25 +7,163 @@
+              task.newListr<TaskContext>(
+                [
+                  {
+                    title: `Obtanining snapshot for baseline '${gitRev}'`,
+                    task: async () => {
+                      ctx.from = await fetchSnapshot(gitRev);
+                    },
+                    retry: {
+                      delay: 2000,
+                      tries: 5,
+                    },
+                  },
+                  {
+                    title: `Taking snapshot of current SO type definitions`,
+                    task: async () => {
+                      ctx.to = await takeSnapshot(ctx.serverHandles!);
+                    },
+                  },
+                ],
+                { concurrent: true }
+              ),
+          },
+          {
+            title: 'Detecting updated types',
+            task: (ctx) => {
+              ctx.updatedTypes = getUpdatedTypes({ from: ctx.from!, to: ctx.to! });
+            },
+          },
+          {
+            title: 'Validating changes in updated types',
+            task: (ctx, task) => {
+              const subtasks: ListrTask<TaskContext>[] = ctx.updatedTypes.map((name) => ({
+                title: `Checking updates on type '${name}'`,
+                task: () => {
+                  try {
+                    validateChanges({
+                      from: ctx.from?.typeDefinitions[name],
+                      to: ctx.to?.typeDefinitions[name]!,
+                    });
+                  } catch (err) {
+                    ctx.invalidUpdates[name] = err;
+                    throw err;
+                  }
+                },
+              }));
+
+              return task.newListr<TaskContext>(subtasks, { exitOnError: false });
+            },
+            skip: (ctx) => ctx.updatedTypes.length === 0,
+          },
+          {
+            title: 'Checking for SO fixtures',
+            task: (ctx, task) => {
+              const registry = ctx.serverHandles?.coreStart.savedObjects.getTypeRegistry();
+              const subtasks = ctx.updatedTypes.map((type) => {
+                return {
+                  title: `Loading fixtures for type '${type}'`,
+                  task: async () => {
+                    const typeFixtures = await getLatestTypeFixtures({
+                      type: registry?.getType(type)!,
+                      snapshot: ctx.to!,
+                      fix,
+                    });
+                    ctx.fixtures[type] = typeFixtures;
+                  },
+                };
+              });
+              return task.newListr<TaskContext>(subtasks, { exitOnError: false });
+            },
+            skip: (ctx) => ctx.updatedTypes.length === 0,
+          },
+          {
+            title: 'Automated rollback tests',
+            task: () => {},
+            skip: (ctx) => Object.keys(ctx.invalidUpdates).length > 0,
+          },
+        ],
+        { fallbackRenderer: 'simple', collectErrors: 'minimal' }
```

Comment:
We might want to explicitly define these to prevent default changes playing havoc with our tests.
```suggestion
        { fallbackRenderer: 'simple', collectErrors: 'minimal', concurrent: `false`, exitOnError: `true` }
```